src/components/table/column-picker/column-picker.component.ts

```diff
@@ -98,74 +101,78 @@ export class ColumnPickerComponent implements OnChanges {
 
     /** Parse data into suitable format for the FlatTreeComponent to understand and initialize deselect tree */
     private rebuildDeselectTree(): void {
```

Comment:
This function is too large and needs to be refactored. Have a read of this, you should have access with your corporate credentials:

https://learning.oreilly.com/library/view/clean-code/9780136083238/chapter03.html

It should look something like
```ts
private rebuildDeselectTree(): void {
  const columns = this.getDeselectedColumnsInOrder();
  this._treeData = this.createTreeData(columns);
  // the rest is fine
}
```

A common practice in Angular is to do data manipulation in an associated service, so that the component is just handling presentation stuff and user input.

Reply:
Most of this code was already here, but with my additions I agree that it's definitely too long. I've moved it into a service as you suggested.

Reply:
Yes, it already needed refactoring before you started, thanks for taking care of it now.